Fix IPC::Connection::sendOutputMessage(IPC::UnixMessage&) Syscall param sendmsg(msg.msg_iov[2]) points to uninitialised byte(s)

jameshilliard opened the pull request:
#### f4367f2cffe45c4514079c73232cf425757ab44e
<pre>
Fix IPC::Connection::sendOutputMessage(IPC::UnixMessage&amp;) Syscall param sendmsg(msg.msg_iov[2]) points to uninitialised byte(s)
<a href="https://bugs.webkit.org/show_bug.cgi?id=242518">https://bugs.webkit.org/show_bug.cgi?id=242518</a>

Reviewed by Fujii Hironori.

Initialize deviceScaleFactor and updateScaleFactor so that we don&apos;t try
to encode uninitialized memory.

* Source/WebKit/Shared/UpdateInfo.h:

Canonical link: <a href="https://commits.webkit.org/252330@main">https://commits.webkit.org/252330@main</a>
</pre>

fujii commented on Source/WebKit/Shared/UpdateInfo.h:
```diff
@@ -52,7 +52,7 @@ class UpdateInfo {
     // The size of the web view.
     WebCore::IntSize viewSize;
-    float deviceScaleFactor;
+    float deviceScaleFactor { 0.0f };
```
WebKit doesn't allow this style.
https://webkit.org/code-style-guidelines/#float-suffixes

jameshilliard replied:
that better?

fujii replied:
See https://webkit.org/code-style-guidelines/#float-suffixes

jameshilliard replied:
I force pushed the PR to use this, was that not right?:
```c
float deviceScaleFactor { 0 };
```

jameshilliard commented on Source/WebKit/Shared/UpdateInfo.h:
```diff
@@ -52,7 +52,7 @@ class UpdateInfo {
     // The size of the web view.
     WebCore::IntSize viewSize;
-    float deviceScaleFactor;
+    float deviceScaleFactor { 0 };
```
Is this not correct?

smfr replied:
Normally we'd initialize a scale factor to 1, not 0.

jameshilliard replied:
Should I change that in a new PR? I thought this was getting set elsewhere if used and just needed to be initialized to anything.

fujii reviewed (approved):
LGTM

jameshilliard committed:
Fix IPC::Connection::sendOutputMessage(IPC::UnixMessage&) Syscall param sendmsg(msg.msg_iov[2]) points to uninitialised byte(s)
https://bugs.webkit.org/show_bug.cgi?id=242518

Reviewed by Fujii Hironori.

Initialize deviceScaleFactor and updateScaleFactor so that we don't try
to encode uninitialized memory.

* Source/WebKit/Shared/UpdateInfo.h:

Canonical link: https://commits.webkit.org/252330@main
